install/migration/migration.py

```diff
@@ -55,7 +55,7 @@ def bind(ldap_uri, base_dn, username, password):
         raise IOError(errno.EIO, 'Cannot get Base DN')
     bind_dn = DN(('uid', username), ('cn', 'users'), ('cn', 'accounts'), base_dn)
     try:
-        conn = ipaldap.LDAPClient(ldap_uri)
+        conn = ipaldap.LDAPClient(ldap_uri, start_tls=True)
```

Comment:
So this may get confusing in the future. The ldap_uri as currently called is api.env.ldap_uri which is an ldapi value (by default) so start_tls doesn't make sense. I believe there are similar changes elsewhere.

Should start_tls be True by default or is that not a backwards-compatible change?

Reply:
I see where it can lead to confusion.

The PR also changes when STARTTLS is performed. The option is now ignored when the protocol is LDAPI or LDAPS.

``start_tls=True`` as default is not fully backward compatible. Windows doesn't support GSSAPI and STARTTLS at the same time.